polynote-frontend/polynote/ui/component/notebooklist.ts

```diff
@@ -392,6 +408,7 @@ export class BranchHandler extends ObjectStateHandler<Branch> {
                 if (!currentState || !isBranch(currentState) || !currentState.children[piece]) {
                     currentUpdate.fullPath = currentPath;
                     currentUpdate.value = piece;
+                    currentUpdate.isOrHasCurrentNotebook = currentNotebook === path;
```

Comment:
does `currentNotebook` have to be updated here? I wonder whether it'd be updated anyways after the path is added (doesn't the state update selecting the current notebook happen after the state update adding a new path? Or am I misremembering?)

Reply:
Just stepping through in devtools, I'm seeing some instances where `branchHandler.addPath` gets called after `branchHandler.updateCurrentNotebook` - not sure where the logic for ordering the state updates lives but it doesn't seem like the behavior is always "add path and then update current notebook."

Reply:
Under what situation do you see that addPath gets called after updateCurrentNotebook? I wasn't able to reproduce. 

The last step of the process to create a new notebook is to select it so I think things should happen in the order I mentioned earlier: https://github.com/polynote/polynote/blob/megan/highlight-current-notebook/polynote-frontend/polynote/messaging/dispatcher.ts#L379

Reply:
I was able to kind of repro when renaming a notebook, I think. But it's hard to tell because we have a bug with that 😅  https://github.com/polynote/polynote/issues/1422

If that's the only way to repro I think we should fix that problem: in general, selecting the current notebook should be the very last thing that happens; and notebook shouldn't be selected before it "exists" if that makes sense.

Reply:
Looks like when reopening the current notebooks (e.g. create a notebook and refresh the page with the notebook still open), the frontend is loading the notebooks and then also getting a message from the server (of type `ListNotebooks`), triggering another state update and causing addPath to get called again.